torchtext/csrc/gpt2_bpe_tokenizer.cpp

```diff
@@ -193,15 +192,15 @@ std::vector<std::string> GPT2BPEEncoder::BPE_(
   //  1) token list size reduced to 1
   //      OR
   //  2) can't find bpe merge
-  auto concatenated = _concatenate_strings(token_list);
+  auto concatenated = concatenate_strings(token_list);
   if (caching_enabled_ && cache_.contains(concatenated)) {
     return cache_.at(concatenated);
   }
 
   std::vector<std::string> tok_list = token_list;
-  auto pairs = _get_pairs(tok_list, seperator_);
+  auto pairs = get_pairs(tok_list, seperator_);
   if (pairs.empty()) {
-    return tok_list;
+    return {concatenated};
```

Comment:
Is this the bug you mentioned previously when trying to use GPT2 BPE for CLIP tokenization?

Reply:
Yes this is the bug. Although GPT-2 never really runs into this because pairs is not empty there. I came across this only when  testing with CLIP tokenizer.